docs/controllers/generics.md

```diff
@@ -0,0 +1,113 @@
+# Generics
+
+This chapter contains tips and tricks for re-using generic reconcilers.
+
+## Generic Reconcilers
+It is possible to create generic reconcilers by abstracing away the underlying type by using either [DynamicObject] or [PartialObjectMeta].
+
+This is a useful technique for controllers that need to do _the same thing_ to a bunch of resources, like for instance adding consistent labels / annotations.
+
+### PartialObjectMeta
+
+This is the easiest and cheapest way to create a generic reconciler because it still retains type information, and you do not have to devolve into a fully dynamic api.
+
+You can create generic reconcilers:
+
+```rust
+pub async fn reconcile<K>(obj: Arc<PartialObjectMeta<K>>, ctx: Arc<Context>)
+-> Result<Action>
+where
+    K: Resource<Scope = NamespaceResourceScope, DynamicType = ()>
+        + Clone
+        + DeserializeOwned
+        + Debug,
+{
+    let kind = K::kind(&()).to_string();
```

Comment:
why do we need this variable? it is not used in this code snipet.
could possibly add a log statement.

Reply:
it's illustrative